File: src/test/java/net/thimmwork/Diagram.java
package net.thimmwork;

import java.util.ArrayList;
import java.util.List;

public class Diagram {
    private final Object[] contents;

    public Diagram(Object[] contents) {
        this.contents = contents;
    }

    public List<String> getLines() {
        List<String> lines = new ArrayList<>(contents.length + 2);
        lines.add("@startuml");
        for (Object it: contents) {
            lines.add(it.toString());
        }
        lines.add("@enduml");
        return lines;
    }
}

File: src/test/java/net/thimmwork/Diagram.kt
package net.thimmwork

import java.util.ArrayList

class Diagram(private val contents: Array<Any>) {

    val lines: List<String>
        get() {
            val lines = ArrayList<String>(contents.size + 2)
            lines += "@startuml"
            contents.forEach { lines += it.toString() }
            lines += "@enduml"
            return lines
        }
}
